pkg/cmd/util/route.go

```diff
@@ -54,10 +57,16 @@ func UnsecuredRoute(kc kclientset.Interface, namespace, routeName, serviceName,
 		},
 	}
 
-	// If the service has multiple ports and the user didn't specify --port,
-	// then default the route port to a service port name.
-	if len(port.Name) > 0 && len(portString) == 0 {
-		route.Spec.Port = resolveRoutePort(port.Name)
+	// When route.Spec.Port is not set, the generator will pick a service port.
+
+	// If the user didn't specify --port, and either the service has a port.Name
+	// or forcePort is set, set route.Spec.Port
+	if (len(port.Name) > 0 || forcePort) && len(portString) == 0 {
+		if forcePort {
+			route.Spec.Port = getTargetPort(svc)
+		} else {
+			route.Spec.Port = resolveRoutePort(port.Name)
```

Comment:
I can remove the call in favor of: 
route.Spec.Port = &api.RoutePort{TargetPort: svc.Spec.Ports[0].TargetPort}

The target.Port is, by default, set to port. So I think it will always be present.
https://kubernetes.io/docs/concepts/services-networking/service/